scripts/release/prepare.sh

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2025 The Jaeger Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script automates the Jaeger release preparation process.
+# It creates a pull request with all the changes necessary for release:
+#   1. Update CHANGELOG.md with the new version and auto-generated release notes.
+#   2. Update the jaeger-ui submodule to the corresponding version
+#   3. Rotate the release managers table in RELEASE.md
+#   4. Create a PR with the 'changelog:skip' label
+#   5. Include exact tag commands in the PR description for post-merge execution.
+#
+# Use:
+#   bash scripts/release/prepare.sh <version>
+#   OR
+#   make prepare-release VERSION=<version>
+#
+# Example:
+#   bash scripts/release/prepare.sh 2.14.0
+#   make prepare-release VERSION=2.14.0
+#
+# After the PR is merged, follow the tag commands in the PR description.
+
+set -euo pipefail
+
+for tool in gh git python3; do
+    if ! command -v "$tool" &> /dev/null; then
+        echo "Error: $tool is not installed or not in PATH"
+        exit 1
+    fi
+done
+
+if [ "$#" -ne 1 ]; then
+    echo "Usage: $0 <version>"
+    echo "Example: $0 2.14.0"
+    exit 1
+fi
+
+VERSION="$1"
+VERSION="${VERSION#v}"
+
+echo "Preparing release for v${VERSION}"
+
+# Verify we are on main branch
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+if [ "$CURRENT_BRANCH" != "main" ]; then
+    echo "Warning: Not on main branch (current: ${CURRENT_BRANCH})"
+    read -p "Continue anyway? (y/n) " -n 1 -r
+    echo
+    if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+        exit 1
+    fi
+fi
+
+git fetch origin
+
+# Create a new branch
+BRANCH_NAME="prepare-release-v${VERSION}"
+git checkout -b "${BRANCH_NAME}"
+
+# Update UI submodule
+echo "Updating UI submodule..."
+git submodule init
+git submodule update
+
+UI_VERSION="v${VERSION}"
+if [ -d "jaeger-ui" ] && [ "$(ls -A jaeger-ui)" ]; then
```

Comment:
not great - if for some reason directory doesn't exist you just silently skip. Should fail instead.